src/services/rename.ts

```diff
@@ -55,6 +55,11 @@ namespace ts.Rename {
         return getRenameInfoSuccess(displayName, fullDisplayName, kind, SymbolDisplay.getSymbolModifiers(typeChecker,symbol), node, sourceFile);
     }
 
+    function isDefinedInLibraryFile(program: Program, declaration: Node) {
+        const sourceFile = declaration.getSourceFile();
+        return sourceFile.hasNoDefaultLib ? false : program.isSourceFileDefaultLibrary(sourceFile);

```

Comment:
Could this always just be `program.isSourceFileDefaultLibrary(sourceFile);`?

Reply:
Actually, it has 

```ts
if (file.hasNoDefaultLib) {
    return true;
}
```

Reply:
@DanielRosenwasser I mean that `isSourceFileDefaultLibrary` return `true` if file has `hasNoDefaultLib`

https://github.com/microsoft/TypeScript/blob/970554733a3578cd854606ebcb316f291632b118/src/compiler/program.ts#L1708-L1712

Maybe we need to ignore JS files instead of checking `hasNoDefaultLib`., what do you think?

Reply:
I think you should check if file is not d.ts file

Reply:
@sheetalkamat Thanks